File: bls_jztc/jztc/src/api/content.ts
import request from '@/utils/request';

// 通用响应结构
export interface ApiResponse<T> {
  code: number;
  message: string;
  data: T;
}

// 内容列表项接口
export interface ContentItem {
  id: number;
  title: string;
  category: string;
  author: string;
  content: string;
  status: string;
  views: number;
  likes: number;
  comments: number;
  isRecommended: boolean;
  publishedAt: string;
  expiresAt: string | null;
  topUntil: string | null;
  createdAt: string;
  updatedAt: string;
}

// 内容列表请求参数
export interface ContentListReq {
  page?: number;
  pageSize?: number;
  title?: string;
  category?: string;
  status?: string;
  author?: string;
}

// 内容列表响应
export interface ContentListRes {
  code: number;
  message: string;
  data: {
    list: ContentItem[];
    total: number;
    page: number;
  };
}

// 内容详情响应
export interface ContentDetailRes {
  code: number;
  message: string;
  data: ContentItem;
}

// 创建内容请求参数
export interface ContentCreateReq {
  title: string;
  category: string;
  author: string;
  content: string;
  status: string;
  isRecommended: boolean;
  topUntil?: string | null;
}

// 创建内容响应
export interface ContentCreateRes {
  code: number;
  message: string;
  data: {
    id: number;
  };
}

// 更新内容请求参数
export interface ContentUpdateReq {
  id: number;
  title: string;
  category: string;
  author: string;
  content: string;
  status: string;
  isRecommended: boolean;
  topUntil?: string | null;
}

// 更新内容响应
export interface ContentUpdateRes {
  code: number;
  message: string;
  data: null;
}

// 删除内容响应
export interface ContentDeleteRes {
  code: number;
  message: string;
  data: null;
}

// 更新内容状态请求参数
export interface ContentStatusUpdateReq {
  id: number;
  status: string;
}

// 更新内容状态响应
export interface ContentStatusUpdateRes {
  code: number;
  message: string;
  data: null;
}

// 更新内容推荐状态请求参数
export interface ContentRecommendUpdateReq {
  id: number;
  isRecommended: boolean;
  topUntil?: string | null;
}

// 更新内容推荐状态响应
export interface ContentRecommendUpdateRes {
  code: number;
  message: string;
  data: null;
}

// 内页轮播图相关接口
// 内页轮播图条目接口
export interface InnerBannerItem {
  id: number;
  image: string;
  linkType: string;
  linkUrl: string;
  isEnabled: boolean;
  order: number;
  bannerType: string;
}

// 更新内页轮播图接口参数
export interface InnerBannerUpdateReq {
  id: number;
  bannerType: string;
  image: string;
  linkType: string;
  linkUrl: string;
  isEnabled: boolean;
  order: number;
}

// 创建内页轮播图请求参数
export interface InnerBannerCreateReq {
  bannerType: string; // 'home' 或 'idle'
  image: string;
  isEnabled: boolean;
  linkType: string;
  linkUrl: string;
  order: number;
}

// 创建内页轮播图
export const createInnerBanner = (data: InnerBannerCreateReq) => {
  return request.post<any, ApiResponse<{ id: number }>>('/content/inner-banner/create', data);
};

// 获取内页轮播图列表
export const getInnerBannerList = (bannerType: string) => {
  return request.get<any, ApiResponse<{ list: InnerBannerItem[], isGlobalEnabled: boolean }>>('/content/inner-banner/list', {
    params: { bannerType }
  });
};

// 更新内页轮播图
export const updateInnerBanner = (data: InnerBannerUpdateReq) => {
  return request.put<any, ApiResponse<null>>('/content/inner-banner/update', data);
};

// 删除内页轮播图
export const deleteInnerBanner = (id: number) => {
  return request.delete<any, ApiResponse<null>>('/content/inner-banner/delete', {
    data: { id }
  });
};

// 更新内页轮播图状态
export const updateInnerBannerStatus = (id: number, isEnabled: boolean) => {
  return request.put<any, ApiResponse<null>>('/content/inner-banner/status/update', {
    id,
    isEnabled
  });
};

// 更新首页轮播图全局开关状态
export const updateHomeInnerBannerGlobalStatus = (isEnabled: boolean) => {
  return request.put<any, ApiResponse<null>>('/content/inner-banner/home/global-status/update', {
    isEnabled
  });
};

// 更新闲置轮播图全局开关状态
export const updateIdleInnerBannerGlobalStatus = (isEnabled: boolean) => {
  return request.put<any, ApiResponse<null>>('/content/inner-banner/idle/global-status/update', {
    isEnabled
  });
};

// 底部Tab相关接口

// 底部Tab项接口定义
export interface BottomTabItem {
  id: number;
  name: string;
  icon: string;
  selectedIcon: string;
  path: string;
  order: number;
  isEnabled: boolean;
}

/**
 * 获取底部Tab列表
 */
export const getBottomTabList = async () => {
  return await request.get<any, ApiResponse<{
    list: BottomTabItem[];
  }>>('/content/bottom-tab/list');
};

/**
 * 更新底部Tab状态
 * @param id Tab ID
 * @param isEnabled 是否启用
 */
export const updateBottomTabStatus = async (id: number, isEnabled: boolean) => {
  return await request.put<any, ApiResponse<null>>('/content/bottom-tab/status/update', {
    id,
    isEnabled
  });
};

/**
 * 更新底部Tab信息
 * @param data Tab数据
 */
export const updateBottomTab = async (data: {
  id: number;
  name: string;
  path: string;
  icon: string;
  selectedIcon: string;
  order: number;
  isEnabled: boolean;
}) => {
  return await request.put<any, ApiResponse<null>>('/content/bottom-tab/update', data);
};

/**
 * 获取内容列表
 * @param params 查询参数
 * @returns 内容列表
 */
export const getContentList = (params: ContentListReq): Promise<ContentListRes> => {
  return request.get('/list', { params });
};

/**
 * 获取内容详情
 * @param id 内容ID
 * @returns 内容详情
 */
export const getContentDetail = (id: number): Promise<ContentDetailRes> => {
  return request.get('/detail', { params: { id } });
};

/**
 * 创建内容
 * @param data 内容数据
 * @returns 创建结果
 */
export const createContent = (data: ContentCreateReq): Promise<ContentCreateRes> => {
  return request({
    url: '/create',
    method: 'post',
    data,
  });
};

/**
 * 更新内容
 * @param data 内容数据
 * @returns 更新结果
 */
export const updateContent = (data: ContentUpdateReq): Promise<ContentUpdateRes> => {
  return request({
    url: '/update',
    method: 'put',
    data,
  });
};

/**
 * 删除内容
 * @param id 内容ID
 * @returns 删除结果
 */
export const deleteContent = (id: number): Promise<ContentDeleteRes> => {
  return request.delete('/delete', { params: { id } });
};

/**
 * 更新内容状态
 * @param data 状态数据
 * @returns 更新结果
 */
export const updateContentStatus = (data: ContentStatusUpdateReq): Promise<ContentStatusUpdateRes> => {
  return request({
    url: '/status/update',
    method: 'put',
    data,
  });
};

/**
 * 更新内容推荐状态
 * @param data 推荐状态数据
 * @returns 更新结果
 */
export const updateContentRecommend = (data: ContentRecommendUpdateReq): Promise<ContentRecommendUpdateRes> => {
  return request({
    url: '/recommend/update',
    method: 'put',
    data,
  });
}; 

File: bls_jztc/demo/api/content/v1/content.go
package v1

import (
	"github.com/gogf/gf/v2/frame/g"
)

// ContentListReq 内容列表请求
type ContentListReq struct {
	g.Meta   `path:"/list" method:"get" tags:"内容管理" summary:"获取内容列表" security:"Bearer" description:"获取内容列表，支持分页和搜索，需要管理员权限"`
	Page     int    `v:"min:1#页码最小值为1" json:"page" dc:"页码"`
	PageSize int    `v:"max:100#每页最大100条" json:"pageSize" dc:"每页数量"`
	Title    string `json:"title" dc:"标题关键词"`
	Category string `json:"category" dc:"分类"`
	Status   string `json:"status" dc:"状态"`
	Author   string `json:"author" dc:"作者"`
}

// ContentListRes 内容列表响应
type ContentListRes struct {
	g.Meta `mime:"application/json" example:"json"`
	List   []ContentListItem `json:"list" dc:"内容列表"`
	Total  int               `json:"total" dc:"总数量"`
	Page   int               `json:"page" dc:"当前页码"`
}

// ContentListItem 内容列表项
type ContentListItem struct {
	Id            int     `json:"id" dc:"内容ID"`
	Title         string  `json:"title" dc:"标题"`
	Category      string  `json:"category" dc:"分类"`
	Author        string  `json:"author" dc:"作者"`
	Status        string  `json:"status" dc:"状态"`
	StatusText    string  `json:"statusText" dc:"状态文本"`
	Views         int     `json:"views" dc:"浏览量"`
	Likes         int     `json:"likes" dc:"想要数量"`
	Comments      int     `json:"comments" dc:"评论数"`
	IsRecommended bool    `json:"isRecommended" dc:"是否推荐"`
	PublishedAt   *string `json:"publishedAt" dc:"发布时间"`
	ExpiresAt     *string `json:"expiresAt" dc:"到期时间"`
	TopUntil      *string `json:"topUntil" dc:"置顶截止时间"`
	CreatedAt     string  `json:"createdAt" dc:"创建时间"`
	UpdatedAt     string  `json:"updatedAt" dc:"更新时间"`
}

// ContentDetailReq 内容详情请求
type ContentDetailReq struct {
	g.Meta `path:"/detail" method:"get" tags:"内容管理" summary:"获取内容详情" security:"Bearer" description:"获取内容详情，需要管理员权限"`
	Id     int `v:"required#内容ID不能为空" json:"id" dc:"内容ID"`
}

// ContentDetailRes 内容详情响应
type ContentDetailRes struct {
	g.Meta        `mime:"application/json" example:"json"`
	Id            int     `json:"id" dc:"内容ID"`
	Title         string  `json:"title" dc:"标题"`
	Category      string  `json:"category" dc:"分类"`
	Author        string  `json:"author" dc:"作者"`
	Content       string  `json:"content" dc:"内容详情"`
	Status        string  `json:"status" dc:"状态"`
	Views         int     `json:"views" dc:"浏览量"`
	Likes         int     `json:"likes" dc:"想要数量"`
	Comments      int     `json:"comments" dc:"评论数"`
	IsRecommended bool    `json:"isRecommended" dc:"是否推荐"`
	PublishedAt   *string `json:"publishedAt" dc:"发布时间"`
	ExpiresAt     *string `json:"expiresAt" dc:"到期时间"`
	TopUntil      *string `json:"topUntil" dc:"置顶截止时间"`
	CreatedAt     string  `json:"createdAt" dc:"创建时间"`
	UpdatedAt     string  `json:"updatedAt" dc:"更新时间"`
	// 闲置物品特有字段
	Type          int     `json:"type" dc:"内容类型 1:普通信息 2:闲置物品"`
	Price         float64 `json:"price" dc:"价格，闲置物品有效"`
	OriginalPrice float64 `json:"originalPrice" dc:"原价，闲置物品有效"`
	TradePlace    string  `json:"tradePlace" dc:"交易地点，闲置物品有效"`
	TradeMethod   string  `json:"tradeMethod" dc:"交易方式，闲置物品有效"`
}

// ContentCreateReq 创建内容请求
type ContentCreateReq struct {
	g.Meta           `path:"/create" method:"post" tags:"内容管理" summary:"创建内容" security:"Bearer" description:"创建新内容，需要管理员权限"`
	Title            string `v:"required#标题不能为空" json:"title" dc:"标题"`
	Category         string `v:"required#分类不能为空" json:"category" dc:"分类"`
	Author           string `v:"required#作者不能为空" json:"author" dc:"作者"`
	Content          string `v:"required#内容不能为空" json:"content" dc:"内容详情"`
	Status           string `v:"required|in:已发布,待审核,已下架#状态不能为空|状态值不正确" json:"status" dc:"状态 已发布/待审核/已下架"`
	IsRecommended    bool   `json:"isRecommended" dc:"是否推荐"`
	ExpiresAt        string `json:"expiresAt" dc:"到期时间"`
	TopUntil         string `json:"topUntil" dc:"置顶截止时间"`
	TopPackageId     int    `json:"topPackageId" dc:"置顶套餐ID，0表示不使用套餐"`
	PublishPackageId int    `json:"publishPackageId" dc:"发布套餐ID，0表示使用默认3天展示期"`
}

// ContentCreateRes 创建内容响应
type ContentCreateRes struct {
	g.Meta         `mime:"application/json" example:"json"`
	Id             int            `json:"id" dc:"内容ID"`
	HomeCategories []CategoryItem `json:"homeCategories" dc:"首页分类列表"`
	IdleCategories []CategoryItem `json:"idleCategories" dc:"闲置分类列表"`
}

// ContentUpdateReq 更新内容请求
type ContentUpdateReq struct {
	g.Meta           `path:"/update" method:"put" tags:"内容管理" summary:"更新内容" security:"Bearer" description:"更新内容信息，需要管理员权限"`
	Id               int    `v:"required#内容ID不能为空" json:"id" dc:"内容ID"`
	Title            string `v:"required#标题不能为空" json:"title" dc:"标题"`
	Category         string `v:"required#分类不能为空" json:"category" dc:"分类"`
	Author           string `v:"required#作者不能为空" json:"author" dc:"作者"`
	Content          string `v:"required#内容不能为空" json:"content" dc:"内容详情"`
	Status           string `v:"required|in:已发布,待审核,已下架#状态不能为空|状态值不正确" json:"status" dc:"状态 已发布/待审核/已下架"`
	IsRecommended    bool   `json:"isRecommended" dc:"是否推荐"`
	ExpiresAt        string `json:"expiresAt" dc:"到期时间"`
	TopUntil         string `json:"topUntil" dc:"置顶截止时间"`
	TopPackageId     int    `json:"topPackageId" dc:"置顶套餐ID，0表示不使用套餐"`
	PublishPackageId int    `json:"publishPackageId" dc:"发布套餐ID，0表示使用默认3天展示期"`
}

// ContentUpdateRes 更新内容响应
type ContentUpdateRes struct {
	g.Meta `mime:"application/json" example:"json"`
}

// ContentDeleteReq 删除内容请求
type ContentDeleteReq struct {
	g.Meta `path:"/delete" method:"delete" tags:"内容管理" summary:"删除内容" security:"Bearer" description:"删除内容，需要管理员权限"`
	Id     int `v:"required#内容ID不能为空" json:"id" dc:"内容ID"`
}

// ContentDeleteRes 删除内容响应
type ContentDeleteRes struct {
	g.Meta `mime:"application/json" example:"json"`
}

// ContentStatusUpdateReq 更新内容状态请求
type ContentStatusUpdateReq struct {
	g.Meta `path:"/status/update" method:"put" tags:"内容管理" summary:"更新内容状态" security:"Bearer" description:"更新内容状态，需要管理员权限"`
	Id     int    `v:"required#内容ID不能为空" json:"id" dc:"内容ID"`
	Status string `v:"required|in:已发布,待审核,已下架#状态不能为空|状态值不正确" json:"status" dc:"状态 已发布/待审核/已下架"`
}

// ContentStatusUpdateRes 更新内容状态响应
type ContentStatusUpdateRes struct {
	g.Meta `mime:"application/json" example:"json"`
}

// ContentRecommendUpdateReq 更新内容推荐状态请求
type ContentRecommendUpdateReq struct {
	g.Meta        `path:"/recommend/update" method:"put" tags:"内容管理" summary:"更新内容推荐状态" security:"Bearer" description:"更新内容推荐状态，需要管理员权限"`
	Id            int    `v:"required#内容ID不能为空" json:"id" dc:"内容ID"`
	IsRecommended bool   `json:"isRecommended" dc:"是否推荐"`
	TopUntil      string `json:"topUntil" dc:"置顶截止时间"`
	TopPackageId  int    `json:"topPackageId" dc:"置顶套餐ID，0表示不使用套餐"`
}

// ContentRecommendUpdateRes 更新内容推荐状态响应
type ContentRecommendUpdateRes struct {
	g.Meta `mime:"application/json" example:"json"`
}

// GetCategoriesReq 获取所有分类请求
type GetCategoriesReq struct {
	g.Meta `path:"/categories" method:"get" tags:"内容管理" summary:"获取所有分类" security:"Bearer" description:"获取所有分类，包括首页分类和闲置分类，需要管理员权限"`
}

// GetCategoriesRes 获取所有分类响应
type GetCategoriesRes struct {
	g.Meta         `mime:"application/json" example:"json"`
	HomeCategories []CategoryItem `json:"homeCategories" dc:"首页分类列表"`
	IdleCategories []CategoryItem `json:"idleCategories" dc:"闲置分类列表"`
}
